spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/query/HqlQueryTransformer.java

```diff
@@ -101,6 +101,10 @@ private static boolean isSubquery(ParserRuleContext ctx) {
 			return false;
 		} else if (ctx instanceof HqlParser.InsertStatementContext) {
 			return false;
+		} else if (ctx instanceof HqlParser.DeleteStatementContext) {
+			return false;
+		} else if (ctx instanceof HqlParser.UpdateStatementContext) {
+				return false;
```

Comment:
We should add another guard in case the parent is empty to eliminate this bug class for future revisions.

Reply:
yeah - sounds good - let me do that